x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/cohere/CohereEmbeddingsRequestEntity.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.cohere;
+
+import org.elasticsearch.inference.InputType;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.cohere.CohereServiceFields;
+import org.elasticsearch.xpack.inference.services.cohere.embeddings.CohereEmbeddingsTaskSettings;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public record CohereEmbeddingsRequestEntity(List<String> input, CohereEmbeddingsTaskSettings taskSettings) implements ToXContentObject {
+
+    private static final String SEARCH_DOCUMENT = "search_document";
+    private static final String SEARCH_QUERY = "search_query";
+    /**
+     * Maps the {@link InputType} to the expected value for cohere for the input_type field in the request
+     */
+    private static final Map<InputType, String> INPUT_TYPE_MAPPING = Map.of(
```

Comment:
Another option would be to do the mapping in the `InputType` enum. I figured that might be tying it to cohere too much since that class might be used for other services. I'm happy to do something else though, it is kind of annoying to do the mapping instead of just using the enum directly.